File: SGCP/SGCP/DTOs/Component/ComponentBuilderDto.cs
namespace SGCP.DTOs.Component
{
    public class ComponentBuilderDto:ComponentCreateDto
    {
        public long Id { get; set; }
        public string Name { get; set; } = string.Empty;
        public string? Code { get; set; }
        public string? Description { get; set; }

        public int? CategoryId { get; set; }
        public int? ComponentTypeId { get; set; }
        public long UnitId { get; set; }
        public string? Type { get; set; }
        public bool Enable { get; set; } = true;
        public decimal UnitCost { get; set; }
        public List<ComponentPresentationDto> Presentations { get; set; } = new();
        public List<ComponentAttributeDto> Attributes { get; set; } = new();
        public List<ComponentTreatmentDto> Treatments { get; set; } = new();
        public List<ComponentProcessDto> Processes { get; set; } = new();
    }
}

File: Client/src/app/features/components/models/component-builder.dto.ts
export interface ComponentBuilderDto {
  id?: number | null;
  name: string;
  code: string;
  description: string;
  categoryId: number;
  componentTypeId: number;
  unitId: number;
  type: string;
  enable: boolean;
  unitCost: number;
  presentations: PresentationDto[];
  attributes: AttributeDto[];
  treatments: TreatmentDto[];
  processes: ProcessDto[];
}

export interface PresentationDto {
  code: string;
  description: string;
  measure: string;
  quantityPerBase: number;
  price: number;
  unitId: number;
  baseUnitCost: number;
  weightGrams: number;
  lengthMeters: number;
  enable: boolean;
}

export interface AttributeDto {
  attributeName: string;
  attributeValue: string;
  enable: boolean;
}

export interface TreatmentDto {
  treatmentTypeId: number;
  extraCost: number;
  enable: boolean;
}

export interface ProcessDto {
  processTypeId: number;
  scopeTypeId: number;
  quantityApplied: number;
  costPerUnit: number;
  date: string;
  notes: string;
  enable: boolean;
}
